proxy/src/schema/registry.rs

```diff
@@ -1,82 +1,149 @@
-use futures::future::Future;
-
 use radicle_registry_client::{
-    ed25519::Pair, ClientT, CreateCheckpointParams, RegisterProjectParams, String32, H256,
+    self as registry, ed25519, message, Client, ClientT, CryptoPair, String32, TransactionExtra,
+    H256,
 };
 
 use crate::schema::error::{Error, ProjectValidation};
 
+/// A container to dissiminate and apply operations on the [`Registry`].
+#[derive(GraphQLObject)]
+pub struct Transaction {
+    /// Unique identifier, in actuality the Hash of the transaction. This handle should be used by
+    /// the API consumer to query state changes of a transaction.
+    pub id: juniper::ID,
+    /// List of operations to be applied to the Registry. Currently limited to exactly one. We use
+    /// a Vec here to accommodate future extensibility.
+    pub messages: Vec<Message>,
+    /// Current state of the transaction.
+    pub state: TransactionState,
+    /// Creation time.
+    pub timestamp: String,
+}
+
+/// Required information to issue a new project registration on the [`Regisry`].
+#[derive(juniper::GraphQLObject)]
+pub struct ProjectRegistration {
+    /// The domain/namespace the project should be registered for.
+    pub domain: String,
```

Comment:
Alternatively, you could use `String32` here. That would require you to define a custom GraphQL scalar but would move these errors to the parsing stage.

Reply:
Will be handled later, I captured it in #99.